Error during make -j install

Hi,

I am currently working on a project involving reinforcement learning in turbulence modeling and would like to use TorchFort for my implementation. I have been following the installation instructions provided on the website and successfully installed the required dependencies, including **libtorch**, **yaml-cpp**, and **pybind11**.

The `cmake` step completed without any errors. However, when running `make`, I encountered the following issue related to the `<filesystem>` header:

**System Details**
OS: OpenSUSE
Compilers: gcc-11, g++ -11, gfortran-11

I ran the following command to configure the build:

```
cmake .. -DCMAKE_C_COMPILER=$CC -DCMAKE_CXX_COMPILER=$CXX -DCMAKE_Fortran_COMPILER=$FC \
-DCMAKE_CXX_STANDARD=17 -DCMAKE_PREFIX_PATH=$CMAKE_PREFIX_PATH \
-DTORCHFORT_YAML_CPP_ROOT=$TORCHFORT_YAML_CPP_ROOT -DTORCHFORT_ENABLE_GPU=OFF
```

This completed successfully. However, during `make -j install`, I encountered multiple errors stating:

```
[  6%] Building CXX object CMakeFiles/torchfort.dir/src/csrc/model_state.cpp.o
[  6%] Building CXX object CMakeFiles/torchfort.dir/src/csrc/distributed.cpp.o
[  9%] Building CXX object CMakeFiles/torchfort.dir/src/csrc/logging.cpp.o
[ 12%] Building CXX object CMakeFiles/torchfort.dir/src/csrc/model_wrapper.cpp.o
[ 15%] Building Fortran object CMakeFiles/torchfort_fort.dir/src/fsrc/torchfort_m.F90.o
[ 18%] Building CXX object CMakeFiles/torchfort.dir/src/csrc/model_pack.cpp.o
[ 21%] Building CXX object CMakeFiles/torchfort.dir/src/csrc/param_map.cpp.o
[ 24%] Building CXX object CMakeFiles/torchfort.dir/src/csrc/setup.cpp.o
/tmp/sachinbm/torch-fort/TorchFort/src/csrc/logging.cpp:31:10: fatal error: filesystem: No such file or directory
 #include <filesystem>
          ^~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/torchfort.dir/build.make:90: CMakeFiles/torchfort.dir/src/csrc/logging.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
/tmp/sachinbm/torch-fort/TorchFort/src/csrc/model_wrapper.cpp:31:10: fatal error: filesystem: No such file or directory
 #include <filesystem>
          ^~~~~~~~~~~~
compilation terminated.
/tmp/sachinbm/torch-fort/TorchFort/src/csrc/model_pack.cpp:31:10: fatal error: filesystem: No such file or directory
 #include <filesystem>
          ^~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/torchfort.dir/build.make:118: CMakeFiles/torchfort.dir/src/csrc/model_wrapper.cpp.o] Error 1
make[2]: *** [CMakeFiles/torchfort.dir/build.make:132: CMakeFiles/torchfort.dir/src/csrc/model_pack.cpp.o] Error 1
In file included from /tmp/sachinbm/torch-fort/TorchFort/src/csrc/model_state.cpp:36:0:
/tmp/sachinbm/torch-fort/TorchFort/src/csrc/include/internal/model_state.h:32:10: fatal error: filesystem: No such file or directory
 #include <filesystem>
          ^~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/torchfort.dir/build.make:104: CMakeFiles/torchfort.dir/src/csrc/model_state.cpp.o] Error 1
In file included from /tmp/sachinbm/torch-fort/TorchFort/src/csrc/setup.cpp:43:0:
/tmp/sachinbm/torch-fort/TorchFort/src/csrc/include/internal/logging.h:33:10: fatal error: filesystem: No such file or directory
 #include <filesystem>
          ^~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/torchfort.dir/build.make:160: CMakeFiles/torchfort.dir/src/csrc/setup.cpp.o] Error 1
[ 27%] Linking Fortran shared library lib/libtorchfort_fort.so
[ 27%] Built target torchfort_fort
make[1]: *** [CMakeFiles/Makefile2:85: CMakeFiles/torchfort.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

It seems that the compiler is not recognizing `<filesystem>`, despite using gcc- 11. Could you please help me resolve this issue? Let me know if you need any additional details.

Looking forward to your response!

I started working with the cartpole example, and when building the example, I got an error as follows:
```
CMake Error at /tmp/sachinbm/torch-fort/pybind11/build/install/share/cmake/pybind11/pybind11NewTools.cmake:300 (if):
  if given arguments:

    "NOT" "ARG_WITHOUT_SOABI" "AND" "NOT" "WITH_SOABI" "IN_LIST" "ARG_UNPARSED_ARGUMENTS"

  Unknown arguments specified
Call Stack (most recent call first):
  CMakeLists.txt:20 (pybind11_add_module)
```

After commenting out those specific lines in pybind11NewTools.cmake, the build files were generated but with some warnings. 

Next `make` gave:
```
/tmp/sachinbm/torch-fort/TorchFort/examples/cpp/cart_pole/train.cpp:34:10: fatal error: yaml-cpp/yaml.h: No such file or directory
   34 | #include <yaml-cpp/yaml.h>
      |          ^~~~~~~~~~~~~~~~~
compilation terminated.
```

Why is error popping? Do I need to set some additional environment variables?

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -34,8 +34,7 @@ endif()
 
 project(torchfort LANGUAGES ${LANGS})
 
-message(STATUS "${CMAKE_CXX_COMPILER_ID}")
-if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+if (CMAKE_CXX_COMPILER_ID STREQUAL "NVHPC")
   # __rdtsc() in torch not supported by nvc++. Use g++ for CXX files.
   message(FATAL_ERROR "TorchFort does not support compilation of C++ files with nvc++. "
                       "Set CMAKE_CXX_COMPILER to g++ to proceed.")

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -25,15 +25,22 @@ if (NOT TORCHFORT_YAML_CPP_ROOT)
   message(FATAL_ERROR "Please set TORCHFORT_YAML_CPP_ROOT to yaml-cpp installation directory.")
 endif()
 
-# __rdtsc() in torch not supported by nvc++. Use g++ for CXX files.
-set(CMAKE_CXX_COMPILER "g++")
 
 if (TORCHFORT_BUILD_FORTRAN)
   set(LANGS Fortran CXX)
 else()
   set(LANGS CXX)
 endif()
+
 project(torchfort LANGUAGES ${LANGS})
+
+message(STATUS "${CMAKE_CXX_COMPILER_ID}")
+if (NOT CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+  # __rdtsc() in torch not supported by nvc++. Use g++ for CXX files.
+  message(FATAL_ERROR "TorchFort does not support compilation of C++ files with nvc++. "
+                      "Set CMAKE_CXX_COMPILER to g++ to proceed.")
+endif()
+
 
 # unit testing with gtest
 if (TORCHFORT_BUILD_TESTS)

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -68,6 +68,7 @@ COPY . /torchfort
 RUN cd /torchfort && mkdir build && cd build && \
     CUDA_PATH=/opt/nvidia/hpc_sdk/Linux_x86_64/24.1/cuda \
     cmake -DCMAKE_INSTALL_PREFIX=/opt/torchfort \
+    -DCMAKE_CXX_COMPILER=`which g++` \
     -DTORCHFORT_YAML_CPP_ROOT=/opt/yaml-cpp \
     -DTORCHFORT_NCCL_ROOT=/opt/nccl/build \
     -DTORCHFORT_BUILD_EXAMPLES=1 \

